src/chat/tools/getEnvironments.ts

```diff
@@ -25,6 +30,40 @@ export interface IGetEnvironmentsParameters {
 export class GetEnvironmentsTool extends BaseLanguageModelTool<IGetEnvironmentsParameters> {
   readonly name = "get_environments";
 
+  prepareInvocation(
+    options: LanguageModelToolInvocationPrepareOptions<IGetEnvironmentsParameters>,
+  ): ProviderResult<PreparedToolInvocation> {
+    const { input } = options;
+    let invocationMessage: string;
+    let confirmationMessage: MarkdownString;
+
+    if (input.connectionId) {
+      invocationMessage = `Get all environments for connection ID: ${input.connectionId}`;
+      confirmationMessage = new MarkdownString()
+        .appendMarkdown(`## Environments Lookup\n`)
+        .appendMarkdown(
+          `This tool will look up all environments associated with connection ID **${input.connectionId}**. Results will show the environment ID and name. Do you want to proceed?`,
```

Comment:
One minor clarification that could be made (this branch or a follow-up) would be that it also includes the names, IDs, and (if CCloud) the cloud provider/region for Kafka clusters, Schema Registries, and Flink compute pools
<img width="360" alt="image" src="https://github.com/user-attachments/assets/986bb19a-b703-4388-a570-4575db6113c3" />